rustls/src/msgs/handshake.rs

```diff
@@ -1214,15 +1224,28 @@ impl Codec for ServerHelloPayload {
         self.session_id.encode(bytes);
         self.cipher_suite.encode(bytes);
         self.compression_method.encode(bytes);
-        codec::encode_vec_u16(bytes, &self.extensions);
+
+        if let Some(ref exts) = self.extensions {
+            codec::encode_vec_u16(bytes, exts);
```

Comment:
Does this do the right thing when `exts` is empty? Or is this assuming that `exts` will never be empty if it is `None`?